parsers/CZ.py

```diff
@@ -0,0 +1,237 @@
+from logging import Logger, getLogger
+from typing import List, Optional, Union
+
+from bs4 import BeautifulSoup
+
+# The request library is used to fetch content through HTTP
+from requests import Response, Session
+
+from parsers.lib.exceptions import ParserException
+
+# please try to write PEP8 compliant code (use a linter). One of PEP8's
+# requirement is to limit your line length to 79 characters.
+
+
+translate_table_gen = {
+    "TPP": "coal",  # coal
+    "CCGT": "gas",  # gas and steem gas
+    "NPP": "nuclear",  # Nuclear
+    "HPP": "hydro",  # Water
+    "PsPP": "hydro",  # Pump Water storage
+    "AltPP": "biomass",  # Alternative
+    "ApPP": "unknown",  # factory
+    "PVPP": "solar",  # photovoltaic
+    "WPP": "wind",  # wind
+    "unknown": "unknown",
+}
+translate_table_dist = {
+    "SEPS": "SVK",
+    "APG": "AT",
+    "PSE": "PL",
+    "TenneT": "DE",
+    "50HzT": "DE",
+}
+
+url = "https://www.ceps.cz/_layouts/CepsData.asmx"
+source = "ceps.cz"
+
+
+def get_mapper(xmlload):
+    series = xmlload.find("series")
+    mapping = {}
+    for tag in series:
+        generator = tag["name"].replace(" [MW]", "")
+        mapping[generator] = tag["id"]
+
+    return mapping
+
+
+def make_request(session, payload, zone_key):
+    headers = {
+        "Content-Type": "application/soap+xml; charset=utf-8",
+        "Content-Length": "1",
+    }
+
+    res: Response = session.post(url, headers=headers, data=payload)
+    assert (
+        res.status_code == 200
+    ), f"Exception when fetching production for {zone_key}: error when calling {url}"
+
+    return res
+
+
+def get_target_datetime(dt: Optional[datetime]) -> datetime:
+    if dt is None:
+        now = datetime.now()
+        dt = (now - timedelta(minutes=now.minute % 15)).replace(second=0, microsecond=0)
+    return dt
+
+
+def __get_exchange_data(
+    zone_key1: str = "CZ",
+    zone_key2: str = "DE",
+    session: Session = Session(),
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+    mode: str = "Actual",
+) -> Union[List[dict], dict]:
+
+    target_datetime = get_target_datetime(target_datetime)
+    from_datetime = target_datetime - timedelta(hours=48)
+
+    payload = """<?xml version="1.0" encoding="utf-8"?>
+    <soap12:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soap12="http://www.w3.org/2003/05/soap-envelope">
+      <soap12:Body>
+        <CrossborderPowerFlows xmlns="https://www.ceps.cz/CepsData/">
+          <dateFrom>{0}</dateFrom>
+          <dateTo>{1}</dateTo>
+          <agregation>{2}</agregation>
+          <function>{3}</function>
+          <version>{4}</version>
+        </CrossborderPowerFlows>
+      </soap12:Body>
+    </soap12:Envelope>""".format(
+        from_datetime.isoformat(), target_datetime.isoformat(), "QH", "AVG", "RT"
+    )
+
+    content = make_request(session, payload, zone_key1).content
```

Comment:
```suggestion
    content = make_request(session, payload, zone_key1).text
```
Won't change anything functionality wise but should satisfy the the type requirement for BeautifulSoup better.